hummingbot/connector/exchange/loopring/loopring_exchange.pyx

```diff
@@ -450,8 +477,14 @@ cdef class LoopringExchange(ExchangeBase):
             }
 
             res = await self.api_request("DELETE", ORDER_CANCEL_ROUTE, params=cancellation_payload, secure=True)
-            if res['resultInfo']['code'] != 0 and res['resultInfo']['message'] != "order in status CANCELLED can't be cancelled":
+            code = res['resultInfo']['code']
+            message = res['resultInfo']['message']
+            if code == 102117:
+                # Order didn't exist on exchange, mark this as canceled
+                self.c_trigger_event(ORDER_CANCELLED_EVENT, cancellation_event)
+            elif code != 0 and (code != 100001 or message != "order in status CANCELLED can't be cancelled"):
                 raise Exception(f"Cancel order returned code {res['resultInfo']['code']} ({res['resultInfo']['message']})")
+
```

Comment:
Orders are only removed from tacking upon confirmation from the exchange that they are in-fact cancelled. The cancel request response only indicates that a cancel request has been scheduled, not that the order has been successfully cancelled. Confirmation of cancellation only happens upon processing the order status updates from either the websocket or the polling. Orders that don't exist are handled carefully as a work around for responses that might indicate that the order doesn't exist when it simply hasn't had time to propagate on the exchange such that the response incorrectly identifies the order as non-existent. They are marked as cancelled appropriately when we can be sure that they are not actually live on the books during the polling.

All of this is based on the exchange's specific intricacies, and we will otherwise miss fills.

Reply:
An extra check to the non-existent order edge case in that function was now added, with a call to stop_tracking as needed